src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -60,8 +60,10 @@ public void EndStreamValueScope()
         /// <returns>A task representing the asynchronous operation. The task result contains a stream for writing the stream value.</returns>
         public async Task<Stream> StartStreamValueScopeAsync()
         {
-            this.WriteSeparatorIfNecessary();
+            this.CommitUtf8JsonWriterContentsToBuffer();
+            this.WriteItemWithSeparatorIfNeeded();

```

Comment:
Why the change from `WriteSeparatorIfNecessary` to `WriteItemWithSeparatorIfNeeded`? Which item would get written in this case?

Reply:
It's still the same separator, it's just that `WriteItemWithSeparatorIfNeeded` has logic responsible for writing a separator before an item if the item is within an array and it is not the first item.